### PR TITLE
Bean name matching logic: OR→AND

### PR DESCRIPTION
**Pull request by yannmh**
`matchBeanName` was incorrectly reporting `true` as soon as one
bean parameter in a configuration was matching and regardless of the
others.
To sum up, fix bean parameters matching logic: `OR`→`AND`

### CODE REVIEW

File: src/test/java/org/datadog/jmxfetch/TestApp.java [hunk 1]
```diff
@@ -332,397 +243,104 @@ public void testExitWatcher() throws Exception {
...
+        assertMetric("this.is.100", 100.0, commonTags, new ArrayList<String>() {{add("foo");add("gorch");add("bar:baz");}} , 8);
+        assertMetric("jmx.org.datadog.jmxfetch.test.number_big", 1.2345678890123457E20, commonTags, 5);
+        assertMetric("jmx.org.datadog.jmxfetch.test.long42424242",4.2424242E7, commonTags, 5);
+        assertMetric("jmx.org.datadog.jmxfetch.test.int424242", 424242.0, commonTags, 5);
+        assertMetric("jmx.org.datadog.jmxfetch.test.should_be1000", 1000.0, commonTags, 5);
+        assertMetric("test.converted", 5.0, commonTags, 5);
+        assertMetric("test.boolean", 1.0, commonTags, 5);
+        assertMetric("test.defaulted", 32.0, commonTags, 5);
+        assertMetric("subattr.this.is.0", 0.0, commonTags, 5);
+        assertMetric("subattr.this.is.0", 0.0, commonTags, 5);
```
Comment (truthbk) [inline src/test/java/org/datadog/jmxfetch/TestApp.java]: Duped line.

File: src/test/java/org/datadog/jmxfetch/TestApp.java [hunk 2]
```diff
@@ -332,397 +243,104 @@ public void testExitWatcher() throws Exception {
...
+        assertMetric("this.is.100", 100.0, commonTags, 8);
+        assertMetric("jmx.org.datadog.jmxfetch.test.number_big", 1.2345678890123457E20, commonTags, 5);
+        assertMetric("jmx.org.datadog.jmxfetch.test.long42424242",4.2424242E7, commonTags, 5);
+        assertMetric("jmx.org.datadog.jmxfetch.test.int424242", 424242.0, commonTags, 5);
+        assertMetric("jmx.org.datadog.jmxfetch.test.should_be1000", 1000.0, commonTags, 5);
+        assertMetric("test.converted", 5.0, commonTags, 5);
+        assertMetric("test.boolean", 1.0, commonTags, 5);
+        assertMetric("test.defaulted", 32.0, commonTags, 5);
+        assertMetric("subattr.this.is.0", 0.0, commonTags, 5);
+        assertMetric("subattr.this.is.0", 0.0, commonTags, 5);
```
Comment (truthbk) [inline src/test/java/org/datadog/jmxfetch/TestApp.java]: Duped line.

File: src/test/java/org/datadog/jmxfetch/TestApp.java [hunk 3]
```diff
@@ -332,397 +243,104 @@ public void testExitWatcher() throws Exception {
...
```
Comment (truthbk) [inline src/test/java/org/datadog/jmxfetch/TestApp.java]: Duped line.

**Comment by yannmh** [comment]
Closing in favor of #81

**Pull request closed without merging**